src/Microsoft.Diagnostics.Monitoring.EventPipe/Triggers/AspNet/AspNetRequestStatusTriggerSettings.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Diagnostics.Monitoring.EventPipe.Triggers.AspNet
+{
+    internal sealed class AspNetRequestStatusTriggerSettings : AspNetTriggerSettings
+    {
+        private const string StatusRegex = "[1-5][0-9]{2}";
+        private static readonly string RangeRegex = FormattableString.Invariant($"^{StatusRegex}(-{StatusRegex})?$");
+
+        /// <summary>
+        /// Specifies the set of status codes for the trigger. This can be individual codes or ranges.
+        /// E.g. 200;400-500
+        /// </summary>
+        [Required]
+        public string[] StatusCodes { get; set; }
```

Comment:
Instead of `string[]`, consider creating some kind of `StatusCodeRange[]` type where a single status code is effectively `[code]` (in range terms) or `[startCode,endCode]` (in range terms). The `StatusCodeRange` struct would understand how to parse a single code or a range of codes. Then the trigger can call some kind of `IsInRange(int value)` method on each `StatusCodeRange` instance.

I think would better separate the concerns of parsing, validation, and range checking from the rest of the code. Again, feel free to defer this.